### PR TITLE
chore(KFLUXVNGD-110): add image-controller

### PR DESCRIPTION
**Pull request by avi-biton**
add image-controller

### CODE REVIEW

File: argo-cd-apps/environments/production/kustomization.yaml [hunk 1]
```diff
@@ -9,7 +9,8 @@ resources:
   - ../../dependencies/pipelines-as-code  
   - ../../dependencies/tekton-config
   - ../../dependencies/tekton-operator
-  - ../../dependencies/tekton-results
```
Comment (yftacherzog) [inline argo-cd-apps/environments/production/kustomization.yaml:12]: There are some trailing whitespaces on some of the lines above. Could you remove them as well?

Comment (yftacherzog) [reply argo-cd-apps/environments/production/kustomization.yaml:12]: Same in staging

**Comment by yftacherzog** [comment]
@avi-biton fetching token and organization will be added in a later PR?

**Comment by avi-biton** [comment]
> @avi-biton fetching token and organization will be added in a later PR?

We are blocked by the lack of `external-secrets`, so as a first step the secret that we create using the org and token will be created manually in the cluster.
Once `external-secrets` will be available for us we will create a resource for this secret in `image-controller` using `ArgoCD`

File: argo-cd-apps/environments/production/kustomization.yaml (hunk 1 above)
Comment (avi-biton) [reply argo-cd-apps/environments/production/kustomization.yaml:12]: We need to setup a CI for this repository, add lint and other tests

Comment (avi-biton) [reply argo-cd-apps/environments/production/kustomization.yaml:12]: done

**Review by yftacherzog — APPROVED** [review]
lgtm

**Pull request merged**